sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -104,3 +110,37 @@ def test_md5sums():
         assert_false(sg.check_md5sum_change(f.name))
 
     os.remove(f.name + '.md5')
+
+
+def test_pattern_matching():
+    """Test if only examples matching pattern are executed"""
```

Comment:
:) ... although sounds more like something to be addressed in nosetests rather than sphinx-gallery. If you don't have docstrings, you don't see the test in the list when you run `nosetests` on the full file / folder.